batch.go

```diff
@@ -0,0 +1,113 @@
+package format
```

Comment:
This is a *lot* of code for an interface package. Unfortunately, the interface needs this (unless we decide to introduce a `Batch` interface.